File: Herança - Subclasses de Personagem/aula4_heranca_JAVA/Mago.java
package aula4_heranca_JAVA;

public class Mago extends Personagem {
  
    private int mana;

    public Mago(String nome, int vida, int nivel){
    super(nome, vida, nivel);
    mana = 50;
}
public int getMana(){
    return mana;
}
public void setMana(int mana){
    if (mana >= 0){
        this.mana = mana;
    } else {
        System.err.println("Mana menor que 0, é inválida!");
    }
}
@Override
public String ficha() {
    return super.ficha() + "\nMana: " + mana;
}
}

File: Polimorfismo - Batalha Final/python/mago.py
from personagem import Personagem

class Mago(Personagem):
    def __init__(self, nome, vida, nivel):
        super().__init__(nome, vida, nivel)
        self.mana = 50

    def get_mana(self):
        return self.mana

    def set_mana(self, mana):
        if mana >= 0:
            self.mana = mana
        else:
            print("Mana menor que 0, é inválida!")

    def ficha(self):
        return super().ficha() + f"\nMana: {self.mana}"

    def habilidade(self):
        return "rajada arcana"